src/Common/Commands.Common/AzurePSCmdlet.cs

```diff
@@ -137,6 +137,39 @@ protected string PSVersion
 
         protected abstract string DataCollectionWarning { get; }
 
+        private SessionState _asJobSessionState;
+
+        public SessionState AsJobSessionState
+        {
+            get
+            {
+                if (_asJobSessionState == null)
+                {
+                    _asJobSessionState = this.SessionState;
+                    return _asJobSessionState;
+                }
+                return _asJobSessionState;
+            }
+            set
+            {
+                _asJobSessionState = value;
+            }
+        }
+
+        private RuntimeDefinedParameterDictionary _copiedDynamicParameters;
```

Comment:
How are we assignign a value to this?

Reply:
We assign it once the dynamic parameters are created in the cmdlet - there is no base property for DynamicParameters, and they can be changed after the AzurePSCmdlet is instantiated.